### PR TITLE
Configuration page for react gui

### PR DESCRIPTION
**Pull request by dimroc**
The placement of the configuration link in the top right is temporary and will be replaced by the story https://www.pivotaltracker.com/story/show/158712734

image

### CODE REVIEW

File: gui/src/containers/Configuration.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  classes: PropTypes.object.isRequired,
+  configs: PropTypes.array.isRequired,
+  fetching: PropTypes.bool.isRequired,
+  error: PropTypes.string
+}
+
+const mapStateToProps = state => {
+  let configError
+  if (state.configuration.networkError) {
+    configError = 'There was an error fetching the jobs. Please reload the page.'
```
Comment (rupurt) [inline gui/src/containers/Configuration.js]: `There was an error fetching the config...`

**Comment by rupurt** [comment]
Now that #384 is up in PR, can you add the config react-static routes [here](https://github.com/smartcontractkit/chainlink/pull/384/files#diff-5ae4fb95fe4674c2a4030dc87795e5c4R15)

File: gui/src/api.js [hunk 2]
```diff
@@ -32,3 +32,5 @@ export const getJobSpec = (id) => request(`/v2/specs/${id}`)
 export const getJobSpecRuns = (id, page, size) => request(`/v2/specs/${id}/runs`, {page: page, size: size})
 
 export const getAccountBalance = () => request('/v2/account_balance')
+
+export const getConfiguration = () => request('/v2/config')
```
Comment (dimroc) [inline gui/src/api.js]: As a champion of ubiquitous language, I would be remiss if I didn't point out the discrepancy between our API URL (`/v2/config`) and our gui url (`/configuration`). Let's consolidate the two to the same spelling. Which one should we use? I'm leaning towards the shorter one: `config`.

File: gui/src/selectors.js [hunk 3]
```diff
@@ -28,3 +28,20 @@ export const jobRunsCountSelector = (state, jobSpecId) => {
   const spec = jobSpecSelector(state, jobSpecId)
   return spec ? spec.runsCount : 0
 }
+
+const camelCaseToUnderscore = str => (
+  str
```
Comment (rupurt) [inline gui/src/selectors.js]: How do you feel about using a library like [`change-case`](https://github.com/blakeembrey/change-case) for this?

Comment (dimroc) [reply gui/src/selectors.js]: Sure, I'm all for it. Didn't know `change-case` existed. I'll add it.

File: gui/src/selectors.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+const changeCase = require('change-case')
```
Comment (rupurt) [inline gui/src/selectors.js]: `require` -> ES6 `import`

```
import { constantCase } from "change-case"
```

**Comment by rupurt** [comment]
🍎

**Comment by rupurt** [comment]
Can you rebase #385 after this is merged

**Comment by dimroc** [comment]
Done

**Pull request merged**